sbin/prepareWorkspace.sh

```diff
@@ -86,10 +87,11 @@ checkoutAndCloneOpenJDKGitRepo() {
   fi
 
   checkoutRequiredCodeToBuild
+  # shellcheck disable=SC2086
   if [ $checkoutRc -ne 0 ]; then
     echo "RETRYWARNING: Checkout required source failed, cleaning workspace and retrying..."
     cd "${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[WORKING_DIR]}"
-    rm -rf "${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[WORKING_DIR]}/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}"
+    rm -rf "${BUILD_CONFIG[WORKSPACE_DIR]:?}/${BUILD_CONFIG[WORKING_DIR]}/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}"
```

Comment:
Why is the ? needed?

Reply:
If `WORKSPACE_DIR` is empty or not set, we may end up starting from the root dir. If `OPENJDK_SOURCE_DIR` is also empty or not set, everything in the root dir could be deleted. `:?` adds in an additional check which will force the script to fail if `WORKSPACE_DIR` is empty

https://github.com/koalaman/shellcheck/wiki/SC2115

The alternative would be doing `:-` with some default value